### PR TITLE
PACT-723 Add the EchoServer to the ctd-omega-editorial-frontend

### PR DESCRIPTION
**Pull request by felixgb**

### CODE REVIEW

File: test/controllers/HomeControllerSpec.scala [hunk 1]
```diff
@@ -21,22 +21,18 @@
...
 import uk.gov.nationalarchives.omega.editorial.controllers.{ HomeController, SessionKeys }
-import uk.gov.nationalarchives.omega.editorial.models.session.Session
+import support.BaseSpec
 
 /** Add your spec here. You can mock out a whole application including requests, plugins etc.
   *
   * For more information, see https://www.playframework.com/documentation/latest/ScalaTestingWithScalaTest
   */
-class HomeControllerSpec extends PlaySpec with GuiceOneAppPerTest with Injecting {
+class HomeControllerSpec extends BaseSpec {
```
Comment (felixgb) [inline test/controllers/HomeControllerSpec.scala:34]: I wanted to not start the JMS echo server in unit tests, so I changed these unit specs to use a version of the application without the StartupModule.

File: app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      endpoint = Endpoint(Some(DirectAddress(HTTP, "localhost", Some(9324))), "elasticmq"),
+      credentials = None,
+      clientId = ClientId("echo_server_1"),
+      None
+    )
+  )
+
+  def startEchoServer: IO[Unit] = {
+    val consumerResource = for {
+      client <- jmsClient
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala:56]: I think we should add some logging here, e.g.
```
 _ <- Resource.eval(logger.info("Starting EchoServer..."))
```

File: app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    requestQueueName,
+                    concurrencyLevel = consumerConcurrencyLevel,
+                    pollingInterval = 50.millis
+                  )
+    } yield consumer
+
+    consumerResource
+      .use(_.handle { (jmsMessage, messageFactory) =>
+        for {
+          requestText <- jmsMessage.asTextF[IO]
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala:67]: We should add some logging here, e.g.
```
_ <- logger.info(s"Echo Server received message: $requestText")
```

File: app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          responseText = s"Echo Server: $requestText"
+          responseMessage <- messageFactory.makeTextMessage(responseText)
+          requestMessageId = jmsMessage.getJMSMessageId.get
+          _ = responseMessage.setJMSCorrelationId(requestMessageId)
```
Comment (rwalpole) [inline app/uk/gov/nationalarchives/omega/editorial/services/EchoServer.scala:71]: We should add some loggin here, e.g.
```
_ <- logger.info(s"Echo Server sending response message: $responseText with correlationId: $requestMessageId") *> IO.unit
```

File: elasticmq.conf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# the include should be done only once, at the beginning of the custom configuration file
+include classpath("application.conf")
+
+queues {
+  request-general { }
+  omega-editorial-web-application-instance-1 { }
+}
```
Comment (rwalpole) [inline elasticmq.conf:7]: We should add a docker-compose.yml file to this project as the aim is for it to become self-contained.

**Review by rwalpole — APPROVED** [review]

**Comment by userjd** [comment]
The application with the echo server does not compile locally. Does the GITHUB_TOKEN need to be included in build.sbt and if so, could this be added to the README?

**Review by userjd — APPROVED** [review]

**Pull request merged**